WordPress/WordPressTest/ContextManagerTests.swift

```diff
@@ -13,7 +13,52 @@ class ContextManagerTests: XCTestCase {
         }
     }
 
-    func testIterativeMigration() throws {
+    func testIterativeMigration130ToLatest() throws {
+        var objectID: NSManagedObjectID? = .none
+
+        // At the time of writing we are at app version 19.9 and model version 140.
+        // At app version 19.0 we were at model version 137.
+        // Iterating back 10 version is more than plenty to cover a real world scenario.
+        try prepareForMigration(withModelName: "WordPress 130") { context in
```

Comment:
Eventually, I'd like to do something dynamic where we pick up `current - x` versions. But in that case we wouldn't be able to write a test like this because obviously if it's dynamic we don't know how the base version differs from the latest.

I do wonder whether we need this level of detail in the test in the first place 🤔 

This new test follows the template of the previous one, but maybe it's enough to run the migrations and make sure they don't crash, then leave testing the details to the ad hoc test for "migration from `x` to `y`?

Anyways, that's something we can tackle in a followup. For the moment I'm just keen on bringing the test suite back to its previous run time.

Reply:
That means that in a couple of months and versions we'll be back to this test making more and more migrations, the more we add new versions to our `xcdatamodel` though…

Though limiting this test to, say, `130–137` instead of `130–current` would be useless (i.e. once we proved that it passed for 130–137 there's no reason this test would ever fail after we add models 138 and later, since by design of `xcdatamodeld` versioning, we never change older models…), and limiting to `(current - 10)–current` as you said would not work because it would be too dynamic as well… So not sure there's an ideal solution there …

I guess we could live with `130–current` for now, and consider revisiting in a year or so when we'd have a lot of new `xcdatamodel` versions piled on top of 137, again increasing the test duration too much.

Maybe we could use [the "Expiring Todo" SwiftLint rule](https://realm.github.io/SwiftLint/expiring_todo.html) to track that and be reminded about it 1 year from now?

Reply:
This change looks good to me. 👍 

@mokagio Just an idea for the dynamic migration approach you mentioned. Instead of figuring out the differences between the old and latest version, and making assertions based on that, we could assert the database file has indeed been migrated to the latest version. For example, at the end of this test, after migration is completed, we could get `NSManagedObjectModel` of the database file, which is implemented [here](https://github.com/wordpress-mobile/WordPress-iOS/blob/69fe3a244e3fab67a2255a69b95d07cf103ef937/WordPress/Classes/Utility/Migrator/CoreDataIterativeMigrator.swift#L35-L46), and assert it is the latest version.